src/adapters.py

```
@@ -14,12 +22,21 @@
 }
 
 
-def _to_int(value: Any, fallback: int) -> int:
+def _require_positive_int(value: Any, *, code: str, detail: str) -> int:
     try:
         out = int(value)
-        return out if out > 0 else fallback
     except (TypeError, ValueError):
-        return fallback
+        raise AdapterValidationError(code, detail)
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)